app/controllers/reports_controller.rb

```diff
@@ -14,6 +14,11 @@ def manufacturer_donations_summary
   def purchases_summary
     @purchases = current_organization.purchases.during(helpers.selected_range)
     @recent_purchases = @purchases.recent.includes(:vendor)
+    @period_supplies = @purchases.sum(:amount_spent_on_period_supplies_cents)
+    @diapers = @purchases.sum(:amount_spent_on_diapers_cents)
+    @adult_incontinence = @purchases.sum(:amount_spent_on_adult_incontinence_cents)
+    @other = @purchases.sum(:amount_spent_on_other_cents)
+    @total_items = @purchases.joins(:line_items).sum(:quantity)
```

Comment:
This is a lot of instance variables. Can we package this up together into a struct, including the previous two variables?

Reply:
This is also a lot of calculations happening here. Probably better to move these into a method in the same place where we define the struct. E.g. something like

```ruby
def purchase_summary(organization, range)
  purchases = organization.purchases.during(helpers.selected_range)
  PurchaseSummary.new(
    purchases: purchases,
    recent_purchases: purchases.recent.includes(:vendor),
    # etc
end
```

Reply:
Done! I changed It by `Purchase.organization_summary_by_dates(current_organization, helpers.selected_range)`